packages-next/fields-document/src/DocumentEditor/slate-types.ts

```diff
@@ -0,0 +1,83 @@
+// it's very important that this file is
+// NOT IMPORTED BY ANYTHING
+// this file contains module augmentation, if it is imported
+// then it will be in the generated declarations which means it will pollute
+// the consumers types if they're already using slate
+// (ideally Slate would use generics instead of module augmentation so this
+// wouldn't be a problem but for now, it is so
+// DO NOT IMPORT THIS FILE
+// )
```

Comment:
```suggestion
//
// ideally Slate would use generics instead of module augmentation so this
// wouldn't be a problem but for now, it is so
// DO NOT IMPORT THIS FILE
```